### PR TITLE
Fixes #1051: replace deprecated CSSTransitionGroup

### LINKED ISSUE #1051: Replace deprecated dependency react-addons-css-transition-group
### Bug report

- __Package version(s)__: 1.16.0
- __Browser and OS versions__: Chrome 57 on OSX (but it doesn't matter)

#### Steps to reproduce

1. Install Blueprint using YARN

#### Actual behavior

Requires the deprecated `react-addons-css-transition-group` package as a peer dependency.

#### Expected behavior

Allows (preferably, recommends) the [react-transition-group](https://github.com/reactjs/react-transition-group) fork, as recommended by the README for [react-addons-css-transition-group](https://www.npmjs.com/package/react-addons-css-transition-group).

### PR DESCRIPTION
**Pull request by benhjames**
#### Checklist

- [x] [Enable CircleCI for your fork](https://circleci.com/add-projects)

#### Changes proposed in this pull request:

Replaces the deprecated use of `react-addons-css-transition-group` with `react-transition-group/CSSTransitionGroup`. There's only one instance of it actually being used, in `Overlay.tsx` - the rest are tooling changes! 🔨 

#### Reviewers should focus on:

Making sure I didn't break the tooling. Seems to all work fine running `gulp` but probably likely I did something stupid!

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @benhjames! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: gulp/util/webpack-config.js [hunk 1]
```diff
@@ -56,7 +56,6 @@ const EXTERNALS = {
     "jquery": "$",
     "moment": "moment",
     "react": "React",
-    "react-addons-css-transition-group": "React.addons.CSSTransitionGroup",
```
Comment (giladgray) [inline gulp/util/webpack-config.js]: this deletion is missing a corresponding addition. i suspect you'll need to add an entry for `react-transition-group`, like you've done everywhere else, so the library is excluded from `core.bundle.js`.

Comment (benhjames) [inline gulp/util/webpack-config.js]: I've added `"react-transition-group": "react-transition-group"` - hopefully this will work! 🤞

**Comment by benhjames** [comment]
<h3>Add missing webpack entry</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/benhjames/blueprint/4/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a> | <a href='https://circleci.com/api/v1/project/benhjames/blueprint/4/artifacts/0//home/ubuntu/blueprint/packages/site-landing/dist/index.html' target='_blank'>landing</a> | <a href='https://circleci.com/api/v1/project/benhjames/blueprint/4/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/benhjames/blueprint/4/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/benhjames/blueprint/4/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

File: gulp/util/webpack-config.js (hunk 1 above)
Comment (benhjames) [inline gulp/util/webpack-config.js]: Hmm, I'm getting `Cannot find module "react-transition-group/CSSTransitionGroup"` on the built artifact - maybe this should be something else? I'm not too familiar with webpack externals!

Comment (giladgray) [inline gulp/util/webpack-config.js]: 😕 but you got a preview comment! 🏅 
revert it so the site will load and we can investigate?

**Comment by benhjames** [comment]
<h3>Revert "Add missing webpack entry"</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/benhjames/blueprint/5/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a> | <a href='https://circleci.com/api/v1/project/benhjames/blueprint/5/artifacts/0//home/ubuntu/blueprint/packages/site-landing/dist/index.html' target='_blank'>landing</a> | <a href='https://circleci.com/api/v1/project/benhjames/blueprint/5/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/benhjames/blueprint/5/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/benhjames/blueprint/5/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

File: gulp/util/webpack-config.js (hunk 1 above)
Comment (benhjames) [inline gulp/util/webpack-config.js]: Reverting seemed to do the trick 😃

Lemme know if this still looks weird and will take a more in-depth look.

**Comment by giladgray** [comment]
@benhjames ok so `react-transition-group` is definitely being bundled in `core.bundle.js`: https://5-90402437-gh.circle-artifacts.com/0/home/ubuntu/blueprint/packages/core/dist/core.bundle.js (cmd+f "var csstransitiongroup = function").

this means `react-transition-group` should appear in the `EXTERNALS` map like i asked originally.
try changing [this line](https://github.com/palantir/blueprint/pull/1063/commits/40d4c3a3cb6176f93005c64f1c31821128e88eef#diff-8f725a5959aac20a243760a4dc664cb2R61) to `"react-transition-group/CSSTransitionGroup": "CSSTransitionGroup"` to map the exact import path to the global variable name (your original value is not a valid JS variable name).

**Comment by benhjames** [comment]
<h3>Add missing webpack external</h3>

Preview: <strong><a href='https://circleci.com/api/v1/project/benhjames/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/site-docs/dist/index.html' target='_blank'>documentation</a> | <a href='https://circleci.com/api/v1/project/benhjames/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/site-landing/dist/index.html' target='_blank'>landing</a> | <a href='https://circleci.com/api/v1/project/benhjames/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/table/preview/index.html' target='_blank'>table</a></strong>
Coverage: <a href='https://circleci.com/api/v1/project/benhjames/blueprint/6/artifacts/0//home/ubuntu/blueprint/packages/core//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>core</a> | <a href='https://circleci.com/api/v1/project/benhjames/blueprint/6/artifacts/0//home/ubuntu/blueprintpackages/datetime//coverage/PhantomJS%202.1.1%20%28Linux%200.0.0%29/index.html' target='_blank'>datetime</a>

**Comment by giladgray** [comment]
@benhjames 💥  looks great! thanks for the contribution 🎩

**Comment by benhjames** [comment]
Thanks @giladgray - the code diff between those two artifacts also helped me to understand what webpack externals are! 👍

**Comment by giladgray** [comment]
"the more you know" 🎓

**Pull request merged**